bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationLocal.java

```diff
@@ -0,0 +1,220 @@
+            }
+        }
+        return addresses;
+    }
+
+    @Override
+    public int getRefreshInterval() {
+        if (!this.useLocalConnection) {
+            return 0;
+        }
+        return super.getRefreshInterval();
+    }
+
+    /**
+     * Returns information whether the device-side incoming passwords are to be shown as properties (and if so: in the
+     * clear or replaced with ***)
+     *
+     * @return Configured value
+     */
+    public DeviceSidePasswordDisplayMode getIncludeDeviceSidePasswordsInProperties() {
+        return this.includeDeviceSidePasswordsInProperties;
+    }
+
+    /**
+     * Should the incoming (intercepted) device-side updates be a strict match to local IP (if provided) or hostname
+     * (fallback)
+     *
+     * @return True - if requiring exact match, False - if IP mismatch is allowed
+     */
+    public boolean getMatchAnyIncomingDeviceIp() {
+        return this.matchAnyIncomingDeviceIp;
+    }
+
+    @Override
+    protected String getExtraFieldDescription() {
+        return String.format(
+                "hostname=%s, localDeviceIP=%s, localDevicePort=%d, connectionMode=%s, useLocalConnection=%s, stubServerPort=%d, stubServerListenAddresses=%s, includeDeviceSidePasswordsInProperties=%s, matchAnyIncomingDeviceIp=%s",
+                getOrDefault(this::getHostname), getOrDefault(this::getLocalDeviceIP), localDevicePort, connectionMode,
+                useLocalConnection, stubServerPort, getOrDefault(this::getStubServerListenAddresses),
+                includeDeviceSidePasswordsInProperties, matchAnyIncomingDeviceIp);
+    }
+
+    @Override
+    protected void validateInternal() throws ArgoConfigurationException {
+        if (hostname.isEmpty()) {
+            throw ArgoConfigurationException.forEmptyRequiredParam(ArgoClimaBindingConstants.PARAMETER_HOSTNAME,
+                    i18nProvider);
+        }
+
+        if (!useLocalConnection && connectionMode == ConnectionMode.LOCAL_CONNECTION) {
+            throw ArgoConfigurationException.forConflictingParams(
+                    ArgoClimaBindingConstants.PARAMETER_USE_LOCAL_CONNECTION, "OFF",
+                    ArgoClimaBindingConstants.PARAMETER_CONNECTION_MODE, ConnectionMode.LOCAL_CONNECTION, i18nProvider);
+        }
+
+        if (getRefreshInterval() == 0 && connectionMode == ConnectionMode.LOCAL_CONNECTION) {
+            throw ArgoConfigurationException.forConflictingParams(ArgoClimaBindingConstants.PARAMETER_REFRESH_INTERNAL,
+                    getRefreshInterval(), ArgoClimaBindingConstants.PARAMETER_CONNECTION_MODE,
+                    ConnectionMode.LOCAL_CONNECTION, i18nProvider);
+        }
+
+        if (localDevicePort < 0 || localDevicePort >= 65536) {
+            throw ArgoConfigurationException.forParamOutOfRange(ArgoClimaBindingConstants.PARAMETER_LOCAL_DEVICE_PORT,
+                    localDevicePort, i18nProvider, 0, 65536);
+        }
+
+        if (stubServerPort < 0 || stubServerPort >= 65536) {
```

Comment:
If thing-types.xml is right:
```
<parameter name="stubServerPort" type="integer" min="0" max="65536" required="false" 
groupName="serverStub">
```
It should be:
```suggestion
        if (stubServerPort < 0 || stubServerPort > 65536) {
```